src/redux/actions/charts/index.js

```diff
@@ -0,0 +1,23 @@
+import resolveChartsData from '../../services/charts';
+
+export const LOADING_CHARTS_DATA = 'LOADING_CHARTS_DATA';
+export const LOAD_CHARTS_DATA = 'LOAD_CHARTS_DATA';
+
+export default function fetchChartsData() {
+    return (dispatch, getState) => {
+        const { isFirstTime } = getState().charts;
+        if (isFirstTime) {
+            dispatch({ type: LOADING_CHARTS_DATA });
+            resolveChartsData()
+                .then((chartData) => {
+                    dispatch({
+                        type: LOAD_CHARTS_DATA,
+                        payload: chartData,
+                    });
```

Comment:
use explicit action like:
START_LOADING_CHARTS
STOP_LOADING_CHARTS
then here:
dispatch LOAD_CHARTS_DATA
dispatch STOP_LOADING_CHARTS
and in the catch:
dispatch STOP_LOADING_CHARTS